src/python/pants/engine/process_test.py

```diff
@@ -36,253 +25,106 @@ def rule_runner() -> RuleRunner:
     return RuleRunner(
         rules=[
             QueryRule(BinaryPaths, [BinaryPathRequest]),
+            QueryRule(ProcessResult, [Process]),
+            QueryRule(FallibleProcessResult, [Process]),
         ],
     )
 
 
-@dataclass(frozen=True)
-class Concatted:
-    value: str
-
-
-@dataclass(frozen=True)
-class BinaryLocation:
-    bin_path: str
-
-    def __post_init__(self):
-        if not os.path.isfile(self.bin_path) or not os.access(self.bin_path, os.X_OK):
-            raise ValueError(f"path {self.bin_path} does not name an existing executable file.")
-
-
-@dataclass(frozen=True)
-class ShellCat:
-    """Wrapper class to show an example of using an auxiliary class (which wraps an executable) to
-    generate an argv instead of doing it all in CatExecutionRequest.
-
-    This can be used to encapsulate operations such as sanitizing command-line arguments which are
-    specific to the executable, which can reduce boilerplate for generating Process instances if the
-    executable is used in different ways across multiple different types of process execution
-    requests.
-    """
-
-    binary_location: BinaryLocation
-
-    @property
-    def bin_path(self):
-        return self.binary_location.bin_path
-
-    def argv_from_snapshot(self, snapshot):
-        cat_file_paths = snapshot.files
-
-        option_like_files = [p for p in cat_file_paths if p.startswith("-")]
-        if option_like_files:
-            raise ValueError(
-                f"invalid file names: '{option_like_files}' look like command-line options"
-            )
-
-        # Add /dev/null to the list of files, so that cat doesn't hang forever if no files are in the
-        # Snapshot.
-        return (self.bin_path, "/dev/null") + tuple(cat_file_paths)
-
-
-@dataclass(frozen=True)
-class CatExecutionRequest:
-    shell_cat: ShellCat
-    path_globs: PathGlobs
-
-
-@rule
-async def cat_files_process_result_concatted(cat_exe_req: CatExecutionRequest) -> Concatted:
-    cat_bin = cat_exe_req.shell_cat
-    cat_files_snapshot = await Get(Snapshot, PathGlobs, cat_exe_req.path_globs)
-    process = Process(
-        argv=cat_bin.argv_from_snapshot(cat_files_snapshot),
-        input_digest=cat_files_snapshot.digest,
-        description="cat some files",
-    )
-    cat_process_result = await Get(ProcessResult, Process, process)
-    return Concatted(cat_process_result.stdout.decode())
-
-
-def cat_stdout_rules():
-    return [cat_files_process_result_concatted, QueryRule(Concatted, (CatExecutionRequest,))]
-
-
-class TestInputFileCreation(TestBase):
-    @classmethod
-    def rules(cls):
-        return (
-            *super().rules(),
-            QueryRule(ProcessResult, (Process,)),
-            QueryRule(FallibleProcessResult, (Process,)),
-        )
-
-    def test_input_file_creation(self):
```

Comment:
This was merged with `test_multiple_file_creation` and `test_file_in_directory_creation` into a single test lower in the file.